sdccc/src/main/java/com/draeger/medical/sdccc/messages/MessageStorage.java

```diff
@@ -322,20 +322,30 @@ protected MessageContent convertMessageToMessageContent(final Message message) {
     private String getSender(final Message message) {
         final CommunicationContext communicationContext = message.getCommunicationContext();
         if (communicationContext == null) {
+            LOG.warn("Encountered message (uuid={}) without a CommunicationContext.", message);
             return UNKNOWN_SENDER;
         }
         final TransportInfo transportInfo = communicationContext.getTransportInfo();
         if (transportInfo == null) {
+            LOG.warn("Encountered message (uuid={}) without a TransportInfo.", message);
             return UNKNOWN_SENDER;
         }
         if (message.getDirection() == CommunicationLog.Direction.INBOUND) {
             final Optional<String> remoteAddress = transportInfo.getRemoteAddress();
             if (remoteAddress.isEmpty()) {
+                LOG.warn("Encountered inbound message (uuid={}) without a remoteAddress.", message);
                 return UNKNOWN_SENDER;
+            } else {
+                return remoteAddress.orElseThrow();
             }
-            return remoteAddress.orElseThrow();
-        } else {
+        } else if (message.getDirection() == CommunicationLog.Direction.OUTBOUND) {
             return DEFAULT_OUTBOUND_SENDER;
+        } else {
+            // NOTE: will invalidate the TestRun
+            LOG.error("Encountered unknown direction {} in message with uuid={}",
```

Comment:
Do a proper invalidation call instead which can be cached by omitting the variable data from the message.

The variable data can then be put into a trace log call.

Reply:
Done.